src/user_profile.cpp

```diff
@@ -136,11 +135,21 @@ bool UserSignature::signFile(string udf_file)
             fprintf(stderr, "Error extracting public key\n");
             return false;
         }
+        
         char *udf = (char *) calloc(buf.st_size, sizeof(char));
         FILE *fp = fopen(udf_file.c_str(), "r");
-        if (UdfToChar(udf, fp, udf_file.c_str(), buf))
-            if (signUdf(sk, buf, udf, signed_message, signed_message_len))     
-                return createSignature(signed_message, signed_message_len);
+        if (udfFromFile(udf, fp, udf_file.c_str(), buf))
+        {
+            signed_message = new unsigned char[crypto_sign_BYTES + buf.st_size];
+            if (crypto_sign(signed_message, &signed_message_len,
+            reinterpret_cast<const unsigned char *>(udf), buf.st_size, sk) != 0)
+            {
+                fprintf(stderr, "Could not sign message\n");
+                delete[] signed_message;
+                return false;
+            }
+            return createSignature(signed_message, signed_message_len);
```

Comment:
You're leaking `signed_message` here. Better capture the output of `createSignature`, then `delete[] signed_message`, and then `return ret`.